### PR TITLE
[PyCDE] Add or/and reduce

### PR DESCRIPTION
**Pull request by Dragon-Git**
- Add and/or reduce for BitsSignal
- Add and reduce for ArraySignal
- Import Reset to pycde

### CODE REVIEW

File: frontends/PyCDE/src/signals.py [hunk 1]
```diff
@@ -462,16 +486,21 @@ def __ne__(self, other):
     from .circt.dialects import hwarith
     return self.__exec_icmp__(other, hwarith.ICmpOp.PRED_NE, "neq")
 
-  # TODO: This class will contain comparison operators (<, >, <=, >=)
-
   def __lt__(self, other):
-    assert False, "Unimplemented"
+    from .circt.dialects import hwarith
```
Comment (teqdruid) [inline frontends/PyCDE/src/signals.py:490]: This seems to be from your other PR. Plz remove it from this one.

**Review by teqdruid — APPROVED** [review]
Thanks!

File: frontends/PyCDE/src/signals.py [hunk 2]
```diff
@@ -272,6 +272,18 @@ def as_uint(self, width: int = None):
     return self._exec_cast(UIntSignal, ir.IntegerType.get_unsigned, width)
 
 
+def And(*items: List[BitVectorSignal]):
+  """Compute a bitwise 'or' of the arguments."""
```
Comment (mortbopet) [inline frontends/PyCDE/src/signals.py:276]: or -> and

File: frontends/PyCDE/src/signals.py [hunk 3]
```diff
@@ -272,6 +272,18 @@ def as_uint(self, width: int = None):
...
+  from .dialects import comb
+  return comb.AndOp(*items)
+
+
+def Or(*items: List[BitVectorSignal]):
+  """Compute a bitwise 'and' of the arguments."""
```
Comment (mortbopet) [inline frontends/PyCDE/src/signals.py:282]: and -> or

**Pull request merged**